crates/bevy_window/src/window.rs

```diff
@@ -86,36 +96,103 @@ pub enum CompositeAlphaMode {
+    /// from the given size and scale it to fit the screen.
+    SizedFullscreen,
+    /// Creates a fullscreen window that uses the maximum supported size.
+    Fullscreen,
 }
 
-use crate::CursorIcon;
-use std::fmt;
-
-use crate::raw_handle::RawHandleWrapper;
+/// Define how a window will be created and how it will behave.
+#[derive(Default, Bundle, Debug, Clone)]
+pub struct WindowBundle {
+    /// Marker component denoting the entity as a window.
+    pub window: Window,
+    /// The cursor of this window.
+    pub cursor: Cursor,
+    /// The position of this window's cursor.
+    pub cursor_position: CursorPosition,
+    /// What presentation mode to give the window.
+    pub present_mode: PresentMode,
+    /// Which fullscreen or windowing mode should be used?
+    pub mode: WindowMode,
+    /// Where the window should be placed.
+    pub position: WindowPosition,
+    /// What resolution the window should have.
+    pub resolution: WindowResolution,
+    /// Stores the title of the window.
+    pub title: WindowTitle,
+    /// Should the window start minimized, maximized, normal?
+    pub state: WindowState,
+    /// How the alpha channel of textures should be handled while compositing.
+    pub composite_alpha_mode: CompositeAlphaMode,
+    /// Which size limits to give the window.
+    pub resize_constraints: WindowResizeConstraints,
+    /// Should the window be resizable?
+    pub resizable: WindowResizable,
+    /// Should the window have decorations enabled?
+    pub decorations: WindowDecorations,
+    /// Should the window be transparent?
+    pub transparency: WindowTransparency,
+    /// Should the window start focused?
+    pub focus: WindowFocus,
+    /// How the window will interact if it is on a web platform rather than OS window.
+    pub canvas: WindowCanvas,
+}
 
-impl fmt::Display for WindowId {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.as_simple().fmt(f)
-    }
+#[derive(WorldQuery)]
+pub struct WindowComponents<'a> {
+    pub entity: Entity,
+    pub window: &'a Window,
+    pub cursor: &'a Cursor,
+    pub cursor_position: &'a CursorPosition,
+    pub present_mode: &'a PresentMode,
+    pub window_mode: &'a WindowMode,
+    pub position: &'a WindowPosition,
+    pub resolution: &'a WindowResolution,
+    pub title: &'a WindowTitle,
+    pub state: &'a WindowState,
+    pub composite_alpha_mode: &'a CompositeAlphaMode,
+    pub resize_constraints: &'a WindowResizeConstraints,
+    pub resizable: &'a WindowResizable,
+    pub decorations: &'a WindowDecorations,
+    pub transparency: &'a WindowTransparency,
+    pub focus: &'a WindowFocus,
+    #[cfg(target_arch = "wasm32")]
+    pub canvas: &'a WindowCanvas,
 }
 
-impl Default for WindowId {
-    fn default() -> Self {
-        WindowId::primary()
-    }
+#[derive(WorldQuery)]
+#[world_query(mutable)]
+pub struct WindowComponentsMut<'a> {
```

Comment:
Ditto on `'static` lifetime.

Reply:
@alice-i-cecile I'm a bit confused on what should be static here, should it be

```
pub struct WindowComponentsMut {
    field: &'static Field,
    ...
}
```
?

Reply:
Figured this out based on other uses.